gensim/models/phrases.py

```diff
@@ -157,19 +162,34 @@ def __str__(self):
             self.__class__.__name__, len(self.vocab), self.min_count,
             self.threshold, self.max_vocab_size)
 
-    @staticmethod
-    def learn_vocab(sentences, max_vocab_size, delimiter=b'_', progress_per=10000):
+    def learn_vocab(self, sentences):
         """Collect unigram/bigram counts from the `sentences` iterable."""
+        if not self.recode_to_utf8 and sentences is not None:
+            try:
+                sentence = list(next(iter(sentences)))
+                if not isinstance(sentence[0], bytes):
+                    self.delimiter = utils.to_unicode(self.delimiter)
+                    self.is_input_bytes = False
+                sentences = it.chain([sentence], sentences)
+            except:
```

Comment:
Using catch-all except blocks is generally a bad idea, since you could end up catching unexpected exceptions. So catch only the specific exception expected here (`StopIteration`?)